src/main/java/frc/robot/subsystems/vision/VisionSubsystem.java

```diff
@@ -1,13 +1,19 @@
 package frc.robot.subsystems.vision;
 
-import edu.wpi.first.wpilibj.Joystick;
-import edu.wpi.first.wpilibj2.command.button.JoystickButton;
-
-public class VisionSubsystem {
-   
-    
-   public void initialize(){
-       
-   }
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+//no need for an update loop like last year's system since everything is immutable, stateless, and delicious!
+public class VisionSubsystem extends SubsystemBase {
```

Comment:
Can you have this inherit from the BitBucketsSubsystem base class and implement the methods? (most will be empty because vision has no motors)

Reply:
Why? Right now the vision subsystem serves no purpose. It'd probably be better to delete it.

Reply:
The vision api designed is probably more performant than anything extending a subsystem: i don't know why i included this class

Reply:
Ok, go ahead and delete vision subsystem. When you pull the latest code into your project, can you update Robot.java to use your vision code when the Robot tells Turret subsystem the degrees to rotate to?

```java
    public void shoot() {
        turretSubsystem.setAzimuth(visionSubsystem.getAngle());
    }
```